unlock-app/src/constants.js

```diff
@@ -46,7 +46,7 @@ export const TRANSACTION_TYPES = {
  *
  * You should not use this directly, instead use the utils/routes.js lockRoute function
  */
-export const LOCK_PATH_NAME_REGEXP = /\/([a-z0-9]+)\/(0x[a-fA-F0-9]{40})(\/([^#]+))?(#(0x[a-fA-F0-9]{40}))?/
+export const LOCK_PATH_NAME_REGEXP = /\/([a-z0-9]+)\/(0x[a-fA-F0-9]{40})(\/([^#]+))?(\/?#(0x[a-fA-F0-9]{40}))?/
```

Comment:
Jumping into a call but this doesn't quite look right to me / it looks like it could be simplified - is the second clause from the end correct?

Reply:
Hey, I just completely refactored the whole thing to use `RegExp` so we can use variable interpolation to make the thing comprehensible as well as correct. PTAL